### DISCUSSION TITLE
Openai API - OpenAIEmbeddings: UnprocessableEntityError

### DISCUSSION THREAD

**Discussion by doomgrave** [oobabooga/text-generation-webui#4876] [Q&A] [answered]
Hi,
Since textgeneration uses the updated Openai API i cannot get the Langchain "OpenAIEmbeddings" to work. 
The only fix I've found is to Downgrade to 0.28.1.

Is there a known solution for this?

### CODE ###
embeddings = OpenAIEmbeddings(base_url=apiUrl,api_key=openai_api_key)
text = "Algoritma is a data science school based in Indonesia and Supertype is a data science consultancy with a distributed team of data and analytics engineers."
doc_embeddings = embeddings.embed_documents([text])

### ERROR ###
Exception has occurred: UnprocessableEntityError
Error code: 422 - {'detail': [{'type': 'string_type', 'loc': ['body', 'input', 'str'], 'msg': 'Input should be a valid string', 'input': [[2149, 87515, 1764, 374, 264, 828, 8198, 2978, 3196, 304, 24922, 323, 6433, 46169, 374, 264, 828, 8198, 74379, 449, 264, 4332, 2128, 315, 828, 323, 28975, 25175, 13]], 'url': 'https://errors.pydantic.dev/2.5/v/string_type'}, {'type': 'string_type', 'loc': ['body', 'input', 'list[str]', 0], 'msg': 'Input should be a valid string', 'input': [2149, 87515, 1764, 374, 264, 828, 8198, 2978, 3196, 304, 24922, 323, 6433, 46169, 374, 264, 828, 8198, 74379, 449, 264, 4332, 2128, 315, 828, 323, 28975, 25175, 13], 'url': 'https://errors.pydantic.dev/2.5/v/string_type'}]}
httpx.HTTPStatusError: Client error '422 Unprocessable Entity' for url 'http://localhost:5001/v1/embeddings'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422

During handling of the above exception, another exception occurred:

  File "C:\PYTHON_SCRIPTS\EXAMPLES\llm_example\00_start.py", line 31, in <module>
    doc_embeddings = embeddings.embed_documents([text])
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
openai.UnprocessableEntityError: Error code: 422 - {'detail': [{'type': 'string_type', 'loc': ['body', 'input', 'str'], 'msg': 'Input should be a valid string', 'input': [[2149, 87515, 1764, 374, 264, 828, 8198, 2978, 3196, 304, 24922, 323, 6433, 46169, 374, 264, 828, 8198, 74379, 449, 264, 4332, 2128, 315, 828, 323, 28975, 25175, 13]], 'url': 'https://errors.pydantic.dev/2.5/v/string_type'}, {'type': 'string_type', 'loc': ['body', 'input', 'list[str]', 0], 'msg': 'Input should be a valid string', 'input': [2149, 87515, 1764, 374, 264, 828, 8198, 2978, 3196, 304, 24922, 323, 6433, 46169, 374, 264, 828, 8198, 74379, 449, 264, 4332, 2128, 315, 828, 323, 28975, 25175, 13], 'url': 'https://errors.pydantic.dev/2.5/v/string_type'}]}

**Comment by michusx** [comment, marked as answer]
Hi.
I'm working with current version of langchain using text-generation-webui Openai API with following definitions:

```
import os
os.environ["OPENAI_API_BASE"] = "http://127.0.0.1:5000/v1"
os.environ["OPENAI_API_KEY"] = "sk-111111111111111111111111111111111111111111111111"

embeddings = OpenAIEmbeddings(
    deployment="your-embeddings-deployment-name",
    model="sentence-transformers/all-mpnet-base-v2",
    tiktoken_enabled=False
)
```
```
...
vectorstore = FAISS.from_documents(documents=splits, embedding=embeddings)
...

```
I have also added:
openai-embedding_model: "sentence-transformers/all-mpnet-base-v2"
in settings.yaml but according to textgen documentation
[https://github.com/oobabooga/text-generation-webui/wiki/12-%E2%80%90-OpenAI-API]
it is its default embeddings model, so it probably should not be necessary...

**Reply by doomgrave** [reply]
PERFECT!  These two settings fixed the problem!!!
{
model="sentence-transformers/all-mpnet-base-v2",
tiktoken_enabled=False
}

By default it try to use 'text-embedding-ada-002' on my version of langchain/openai.
The flag tiktoken_enabled is hidden anyway.
